src/utils.js

```diff
@@ -4,27 +4,21 @@ function isAsyncIterable (obj) {
   return typeof obj?.[Symbol.asyncIterator] === 'function' || typeof obj?.[Symbol.iterator] === 'function'
 }
 
-function binarySearch (list, target) {
-  if (list.length % 2 !== 0) throw Error('List length must be even')
+function binarySearch (arr, val) {
```

Comment:
When I got around to reading through the algorithm this was based off of (https://www.tbray.org/ongoing/When/200x/2003/03/22/Binary) I couldn't actually explain or assert it was working as expected.

Could you add an explanation how this works for the flattened array of header name/value pairs?

Reply:
It actually bypasses the flattened stuff by just dividing by 2 and treating it as an array of only names. It's just a regular binary search at this point.

Reply:
i'm not sure I understand. 

```js
const headers = [ 'h1', 'v1', 'h2', 'v2', 'h3', 'v3' ]
```

Would only search over the lower half of this array miss `'h3'`?

Reply:
This change passes the test suite so I'm confident its a valid solution; just looking for better explanation

Reply:
I think this solution is the simplest and easiest to understand. I can’t really explain it any better than the code.

Reply:
Can you merge the suggestion comment I just added?